torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -423,6 +432,178 @@ TensorView* TensorView::computeAt(
+  TORCH_INTERNAL_ASSERT(
+      !container()->isA<kir::Kernel>(),
+      "Function invalid for kernel container.");
+
+  if (isFusionInput()) {
+    return;
+  }
+
+  TORCH_CHECK(
+      !ir_utils::consumerTvsOf(this).empty(),
+      "There must be at least one consumer of this tensor to use computeWith: ",
+      toString());
+
+  if (pos < 0) {
+    pos += int(nDims()) + 1;
+  }
+
+  TORCH_INTERNAL_ASSERT(
+      pos >= 0 && pos <= nDims(),
+      "Invalid inline position for ",
+      toString(),
+      ": ",
+      pos);
+
+  const auto max_inline_pos =
+      MaxPosCalculator({}, true).getMaxPosAll(this, best_effort);
+
+  if (best_effort) {
+    pos = std::min<int>(max_inline_pos, pos);
+  }
+
+  // hoist inner most broadcast
+  while (pos > 0 && axis(pos - 1)->isBroadcast()) {
+    pos--;
+  }
+
+  TORCH_CHECK(
+      pos <= max_inline_pos,
+      "Invalid computeWith position for T",
+      name(),
+      ": ",
+      pos,
+      ". Maximum allowed value:",
+      max_inline_pos);
+
+  // The position must be right of the computeAt position
+  TORCH_CHECK(
+      pos >= getComputeAtPosition(),
+      "Position must be right of the computeAt position. Position: ",
+      pos,
+      ", computeAt position: ",
+      getComputeAtPosition());
+
+  // If it's already set to be computed with the consumer and the
+  // position is higher, nothing to change
+  if (getComputeWithPosition() >= pos) {
+    return;
+  }
+
+  // Update the siblings together
+  auto siblings = ir_utils::filterByType<TensorView>(definition()->outputs());
+
+  for (auto sibling : siblings) {
+    sibling->clearComputeWith();
+  }
+
+  // If the given position is the same as the computeAt position, this
+  // is a no-op
+  if (pos == getComputeAtPosition()) {
+    return;
+  }
+
+  for (auto sibling : siblings) {
+    sibling->compute_with_pos_ = (unsigned int)pos;
+  }
+}
+
+bool TensorView::isComputedWith(const TensorView* consumer) const {
+  if (!hasComputeWith()) {
+    return false;
+  }
+
+  // Quering is an error if the compute-with consumer is still unresolved
+  TORCH_INTERNAL_ASSERT(
+      hasResolvedComputeWith(), "Not resolved yet: ", toString());
+
+  return std::find(
+             getComputeWithConsumers().begin(),
+             getComputeWithConsumers().end(),
+             consumer) != getComputeWithConsumers().end();
+}
+
+const std::vector<TensorView*>& TensorView::getComputeWithConsumers() const {
+  TORCH_INTERNAL_ASSERT(
+      !hasComputeWith() || hasResolvedComputeWith(),
+      "computeWith not yet resolved: ",
+      toString());
+  return compute_with_consumers_;
+}
+
+unsigned int TensorView::getMaybeComputeAtPosition(
+    const TensorView* consumer) const {
+  if (hasResolvedComputeWith() && isComputedWith(consumer)) {
+    return getComputeWithPosition();
+  } else {
+    return getComputeAtPosition();
+  }
+}
+
+bool TensorView::resolveComputeWith(const std::vector<Expr*>& sorted_exprs) {
```

Comment:
Sorted expressions is post expr sorting, not just the topologically sorted DAG, correct?